kubernetes/samples/scripts/domain-lifecycle/waitForDomain.sh

```diff
@@ -0,0 +1,598 @@
+          esac
+          ;;
+
+      -p) case "$val" in
+            0|none) EXPECTED_STATE='0' ;;
+            completed|Completed|COMPLETED) EXPECTED_STATE='Completed' ;;
+            *) syntaxError '-p requires value "0" or "Completed"' ;;
+          esac
+          ;;
+
+      -q) VERBOSE=false
+          REPORT_INTERVAL_SECS=30
+          ;;
+
+      -i) IGNORE_FAILURES=true
+          ;;
+
+      '-?') usage
+          exit 0
+          ;;
+    esac
+  done
+}
+
+sortlist() {
+  # sort a comma or space separated list
+  #   - stdin input, stdout output
+  #   - spaces replaced with commas
+  #   - blank fields ignored
+  #   - output removes any trailing comma
+  #   - examples: ""->"" "c,b"->"b,c" "c,b,"->"b,c" "c b"->"b,c"
+  tr ' ' '\n'    | \
+  tr ',' '\n'    | \
+  sort -V        | \
+  xargs echo -n  | \
+  tr ' ' ','
+}
+sortAIImages() {
+  # sort "aiimages=;im2,im1;" field assuming comma or space sep list
+  #   - stdin input, stdout output
+  #   - spaces replaced with commas
+  #   - input ignores trailing comma, output removes any trailing comma
+  #   - examples: see sortAIImagesUnitTest()
+  while read line
+  do
+    echo -n "$line" | sed 's/\(^.*aiimages=;\).*/\1/'
+    echo -n "$line" | sed 's/.*aiimages=;\([^;]*\).*/\1/' | sortlist
+    echo "$line"    | sed 's/.*aiimages=;[^;]*\(;.*\)/\1/'
+  done
+}
+_sortAIImagesUnitTest() {
+  local res=$(echo "$1" | sortAIImages)
+  if [ ! "$res" = "$2" ]; then
+    echo "unit test fail"
+    echo " input ='$1'"
+    echo " expect='$2'"
+    echo " actual='$res'"
+    exit 1
+  fi
+}
+sortAIImagesUnitTest() {
+  _sortAIImagesUnitTest "foo=;bar; aiimages=;c,b; bar=;foo;"   "foo=;bar; aiimages=;b,c; bar=;foo;"
+  _sortAIImagesUnitTest "foo=;bar; aiimages=; c,b,; bar=;foo;" "foo=;bar; aiimages=;b,c; bar=;foo;"
+  _sortAIImagesUnitTest "foo=;bar; aiimages=;; bar=;foo;"      "foo=;bar; aiimages=;; bar=;foo;"
+  _sortAIImagesUnitTest "foo=;bar; aiimages=;a ; bar=;foo;"    "foo=;bar; aiimages=;a; bar=;foo;"
+  _sortAIImagesUnitTest "aiimages=;c b; bar=;foo; foo=;bar;"   "aiimages=;b,c; bar=;foo; foo=;bar;"
+  _sortAIImagesUnitTest "bar=;foo; foo=;bar; aiimages=; c b ;" "bar=;foo; foo=;bar; aiimages=;b,c;"
+  _sortAIImagesUnitTest "aiimages=;;"                          "aiimages=;;"
+  _sortAIImagesUnitTest "aiimages=; ;"                         "aiimages=;;"
+  _sortAIImagesUnitTest "aiimages=;,,;"                        "aiimages=;;"
+  return 0
+}
+sortAIImagesUnitTest
+
+
+getDomainValue() {
+  # get domain value specified by $2 and put in env var named by $1
+  #   - if get fails, and global EXPECTED_STATE is "Completed", then echo an Error and exit script non-zero
+  #   - example: getDomainValue DOM_VERSION '.spec.introspectVersion'
+  local __retvar=$1
+  local ljpath="{$2}"
+  local attvalue
+  set +e
+  attvalue=$(${KUBERNETES_CLI} -n ${DOMAIN_NAMESPACE} get domain ${DOMAIN_UID} -o=jsonpath="$ljpath" 2>&1)
```

Comment:
Since this script only works with `v9+` version of the domain, should we explicitly get the `v9` version of the domain using `${KUBERNETES_CLI} get domain.v9.weblogic.oracle` ?

Reply:
Will it continue to work for future versions of the domain in that case? (The script already explicitly checks whether the version is 9 or higher.)

Reply:
Yes, I believe we will continue to support the v9 version in the future operator version. I made a similar change to `wl-pod-wait.sh` in release/3.4 branch to explicitly use the `v8` version.

Reply:
Done.